Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -16,44 +16,41 @@
 
 #include "Starboard.h"
 #import <StubReturn.h>
+#include <vector>
 
-#include "Foundation/NSOrderedSet.h"
+#import <Foundation/NSOrderedSet.h>
+#import <Foundation/NSException.h>
+
+@implementation NSOrderedSet {
+    std::vector<StrongId<id>> setContainer;
```

Comment:
okay, if I use the map<id,vector<StrongId<id>>::Iterator>
the key is the object and the value is an iterator to the object?
Wouldn't it be simpler to just do this, map<id,NSUInteger> and during the insert call.

//Check if it's contained in the map
vector.push_back(obj);
map[obj] = (vector.size() -1);

this way we are keeping track of the order/index of the object in the map via value.